<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Costa, W. G." />

<meta name="date" content="2025-10-16" />

<title>Processamento de Dados Climáticos para Modelagem Preditiva</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NIR and Genomic Kernel Integration</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="analysis.html">Análise</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wevertongomescosta/Integrating-nir-genomic-kernel">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Processamento de Dados Climáticos para
Modelagem Preditiva</h1>
<h4 class="author">Costa, W. G.<a href="#fn1" class="footnote-ref"
id="fnref1"><sup>1</sup></a></h4>
<h4 class="date">2025-10-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>Integrating-nir-genomic-kernel/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250829code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250829)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250829code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250829)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaIntegratingnirgenomickerneltree6eb368f01f4ec56a3881993ec150fa7b3b43a648targetblank6eb368fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/WevertonGomesCosta/Integrating-nir-genomic-kernel/tree/6eb368f01f4ec56a3881993ec150fa7b3b43a648" target="_blank">6eb368f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaIntegratingnirgenomickerneltree6eb368f01f4ec56a3881993ec150fa7b3b43a648targetblank6eb368fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/WevertonGomesCosta/Integrating-nir-genomic-kernel/tree/6eb368f01f4ec56a3881993ec150fa7b3b43a648" target="_blank">6eb368f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/analysis.Rmd
    Untracked:  analysis/climate_data.Rmd
    Untracked:  analysis/code_NIRS_kernels.R
    Untracked:  analysis/code_for_GBLUP.R
    Untracked:  analysis/code_for_GBLUP_pt.R
    Untracked:  analysis/process_climate_data.Rmd
    Untracked:  data/Article_documents/
    Untracked:  data/GAPIT.Genotype.Numerical.txt
    Untracked:  data/Maize-NIRS-GBS-main/
    Untracked:  data/NIR.csv
    Untracked:  output/GAK.rds
    Untracked:  output/GAKE.rds
    Untracked:  output/GGK.rds
    Untracked:  output/GGKE.rds
    Untracked:  output/PAK.rds
    Untracked:  output/PAKE.rds
    Untracked:  output/PGK.rds
    Untracked:  output/PGKE.rds
    Untracked:  output/Pedigree.rds
    Untracked:  output/Pheno.rds
    Untracked:  output/Pred.ability.CV00.CV0.csv
    Untracked:  output/Pred.ability.CV1.CV2.CV0.CV00.classified.csv
    Untracked:  output/Pred.ability.CV1.CV2.CV0.CV00.csv
    Untracked:  output/Pred.ability.CV1.CV2.csv
    Untracked:  output/Pred.ability.jpeg
    Untracked:  output/Pred.ability.top5.jpeg
    Untracked:  output/ZE.rds
    Untracked:  output/ZG.rds
    Untracked:  output/ZGZE.rds
    Untracked:  output/ZP.rds
    Untracked:  output/ZPZE.rds
    Untracked:  output/environmental_covariates.csv
    Untracked:  output/rep_1/
    Untracked:  output/rep_2/
    Untracked:  output/rep_3/
    Untracked:  output/rep_4/
    Untracked:  output/rep_5/
    Untracked:  output/results/

Unstaged changes:
    Deleted:    LICENSE
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    data/Articles/.gitignore
    Deleted:    data/Articles/The Plant Genome - 2024 - DeSalvio - Near‐infrared reflectance spectroscopy phenomic prediction can perform similarly to.pdf
    Deleted:    data/Y7.csv
    Deleted:    data/YCV1_CV2.csv
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introdução" class="section level2">
<h2>1. Introdução</h2>
<p>Este script busca dados climáticos diários da API NASA POWER para a
localidade de College Station, TX, durante as safras de 2011 e 2012. O
objetivo é transformar os dados diários em <strong>Covariáveis
Ambientais (Environmental Covariates - ECs)</strong> que resumem as
condições de cada ambiente (ano x manejo hídrico).</p>
<p>Essas ECs serão salvas em um arquivo <code>.csv</code> para serem
usadas no script principal de análise (<code>analysis.Rmd</code>) para
modelar a interação GxE.</p>
</div>
<div id="configuração-do-ambiente" class="section level2">
<h2>2. Configuração do Ambiente</h2>
<p>Carregamos os pacotes necessários para a aquisição e manipulação dos
dados.</p>
<pre class="r"><code># Instale se necessário
# install.packages(c(&quot;nasapower&quot;, &quot;dplyr&quot;, &quot;lubridate&quot;, &quot;tidyr&quot;))

library(nasapower)
library(dplyr)
library(lubridate)
library(tidyr)</code></pre>
</div>
<div id="aquisição-dos-dados-climáticos-diários" class="section level2">
<h2>3. Aquisição dos Dados Climáticos Diários</h2>
<p>Buscamos os dados para a estação de crescimento (Maio a Setembro)
para os anos de 2011 e 2012.</p>
<pre class="r"><code># Coordenadas de College Station, TX
coords_texas &lt;- c(lon = -96.3344, lat = 30.6282)

# Parâmetros climáticos de interesse
params &lt;- c(&quot;PRECTOTCORR&quot;, &quot;T2M_MAX&quot;, &quot;T2M_MIN&quot;, &quot;T2M&quot;, &quot;RH2M&quot;)

# Obtém dados para 2011 e 2012
daily_climate_data &lt;- get_power(
  community = &quot;AG&quot;,
  lonlat = coords_texas,
  pars = params,
  dates = c(&quot;2011-05-01&quot;, &quot;2012-09-30&quot;),
  temporal_api = &quot;DAILY&quot;
) %&gt;% 
  # Converte a data e extrai o ano
  mutate(date = ymd(YYYYMMDD),
         year = year(date))

head(daily_climate_data)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["LON"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["LAT"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["YEAR"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["MM"],"name":[4],"type":["int"],"align":["right"]},{"label":["DD"],"name":[5],"type":["int"],"align":["right"]},{"label":["DOY"],"name":[6],"type":["int"],"align":["right"]},{"label":["YYYYMMDD"],"name":[7],"type":["date"],"align":["right"]},{"label":["PRECTOTCORR"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["T2M_MAX"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["T2M_MIN"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["T2M"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["RH2M"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["date"],"name":[13],"type":["date"],"align":["right"]},{"label":["year"],"name":[14],"type":["dbl"],"align":["right"]}],"data":[{"1":"-96.3344","2":"30.6282","3":"2011","4":"5","5":"1","6":"121","7":"2011-05-01","8":"0.00","9":"33.95","10":"16.93","11":"26.85","12":"66.05","13":"2011-05-01","14":"2011"},{"1":"-96.3344","2":"30.6282","3":"2011","4":"5","5":"2","6":"122","7":"2011-05-02","8":"0.99","9":"20.62","10":"11.77","11":"15.45","12":"69.83","13":"2011-05-02","14":"2011"},{"1":"-96.3344","2":"30.6282","3":"2011","4":"5","5":"3","6":"123","7":"2011-05-03","8":"0.14","9":"25.13","10":"10.30","11":"16.68","12":"50.50","13":"2011-05-03","14":"2011"},{"1":"-96.3344","2":"30.6282","3":"2011","4":"5","5":"4","6":"124","7":"2011-05-04","8":"0.00","9":"27.91","10":"9.52","11":"19.04","12":"36.35","13":"2011-05-04","14":"2011"},{"1":"-96.3344","2":"30.6282","3":"2011","4":"5","5":"5","6":"125","7":"2011-05-05","8":"0.01","9":"30.96","10":"12.09","11":"21.34","12":"38.21","13":"2011-05-05","14":"2011"},{"1":"-96.3344","2":"30.6282","3":"2011","4":"5","5":"6","6":"126","7":"2011-05-06","8":"0.00","9":"32.76","10":"13.95","11":"23.84","12":"50.22","13":"2011-05-06","14":"2011"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="engenharia-de-atributos-criação-das-covariáveis-ambientais-ecs"
class="section level2">
<h2>4. Engenharia de Atributos: Criação das Covariáveis Ambientais
(ECs)</h2>
<p>Agora, agregamos os dados diários para criar variáveis que descrevam
cada ano/ambiente.</p>
<p><strong>Definições:</strong> * <strong>GDD (Growing Degree
Days):</strong> Usamos a fórmula padrão para milho (Temperatura Base =
10°C, Temperatura Máxima = 30°C). * <strong>Stress Hídrico:</strong> O
número de dias com precipitação zero. * <strong>Stress Térmico:</strong>
O número de dias com temperatura máxima acima de 35°C.</p>
<pre class="r"><code># Calcula GDD diário
daily_climate_data &lt;- daily_climate_data %&gt;%
  mutate(
    T_base = 10,
    T_upper = 30,
    T_eff_min = if_else(T2M_MIN &lt; T_base, T_base, T2M_MIN),
    T_eff_max = if_else(T2M_MAX &gt; T_upper, T_upper, T2M_MAX),
    T_avg_eff = (T_eff_min + T_eff_max) / 2,
    GDD = T_avg_eff - T_base
  )

# Agrega os dados por ano para criar as ECs
environmental_covariates &lt;- daily_climate_data %&gt;%
  group_by(year) %&gt;%
  summarise(
    # Covariáveis de Temperatura
    TMAX_AVG = mean(T2M_MAX, na.rm = TRUE),
    TMIN_AVG = mean(T2M_MIN, na.rm = TRUE),
    GDD_CUM = sum(GDD, na.rm = TRUE),
    HEAT_STRESS_DAYS = sum(T2M_MAX &gt; 35, na.rm = TRUE),
    
    # Covariáveis Hídricas
    PRECTOT = sum(PRECTOTCORR, na.rm = TRUE),
    DRY_DAYS = sum(PRECTOTCORR == 0, na.rm = TRUE),
    RH_AVG = mean(RH2M, na.rm = TRUE),
    .groups = &#39;drop&#39;
  )

# Replica as linhas para cada combinação de manejo hídrico (WS e WW)
# No seu modelo, a diferença WS/WW já é capturada pelo efeito de ambiente (E)
# As ECs representam o &quot;macro-ambiente&quot; (ano).
ecs_final &lt;- environmental_covariates %&gt;%
  slice(rep(1:n(), each = 2)) %&gt;%
  mutate(water_regime = rep(c(&quot;WS&quot;, &quot;WW&quot;), times = n() / 2),
         Env = paste0(&quot;CS&quot;, substr(year, 3, 4), &quot;_&quot;, water_regime)) %&gt;%
  select(Env, everything(), -water_regime, -year) # Reordena colunas

# Visualiza as covariáveis finais
print(ecs_final)</code></pre>
<pre><code># A tibble: 4 × 8
  Env     TMAX_AVG TMIN_AVG GDD_CUM HEAT_STRESS_DAYS PRECTOT DRY_DAYS RH_AVG
  &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;            &lt;int&gt;   &lt;dbl&gt;    &lt;int&gt;  &lt;dbl&gt;
1 CS11_WS     32.1     18.3   3231.              120    401.      117   58.3
2 CS11_WW     32.1     18.3   3231.              120    401.      117   58.3
3 CS12_WS     28.5     16.9   3285.               64    900.       92   74.1
4 CS12_WW     28.5     16.9   3285.               64    900.       92   74.1</code></pre>
</div>
<div id="salvar-as-covariáveis" class="section level2">
<h2>5. Salvar as Covariáveis</h2>
<p>Salvamos o dataframe final em um arquivo CSV para ser carregado no
script principal.</p>
<pre class="r"><code># Garante que a pasta &#39;output&#39; exista
if (!dir.exists(&quot;output&quot;)) dir.create(&quot;output&quot;)

# Salva o arquivo
write.csv(ecs_final, &quot;output/environmental_covariates.csv&quot;, row.names = FALSE)

cat(&quot;Arquivo &#39;environmental_covariates.csv&#39; salvo com sucesso na pasta &#39;output&#39;!&quot;)</code></pre>
<pre><code>Arquivo &#39;environmental_covariates.csv&#39; salvo com sucesso na pasta &#39;output&#39;!</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=Portuguese_Brazil.utf8  LC_CTYPE=Portuguese_Brazil.utf8   
[3] LC_MONETARY=Portuguese_Brazil.utf8 LC_NUMERIC=C                      
[5] LC_TIME=Portuguese_Brazil.utf8    

time zone: America/Sao_Paulo
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tidyr_1.3.1     lubridate_1.9.4 dplyr_1.1.4     nasapower_4.2.5
[5] workflowr_1.7.2

loaded via a namespace (and not attached):
 [1] utf8_1.2.6        sass_0.4.10       generics_0.1.4    stringi_1.8.7    
 [5] httpcode_0.3.0    hms_1.1.3         digest_0.6.37     magrittr_2.0.4   
 [9] evaluate_1.0.5    timechange_0.3.0  fastmap_1.2.0     rprojroot_2.1.1  
[13] jsonlite_2.0.0    processx_3.8.6    whisker_0.4.1     crul_1.6.0       
[17] ps_1.9.1          urltools_1.7.3.1  promises_1.3.3    httr_1.4.7       
[21] purrr_1.1.0       jquerylib_0.1.4   cli_3.6.5         crayon_1.5.3     
[25] rlang_1.1.6       triebeard_0.4.1   bit64_4.6.0-1     withr_3.0.2      
[29] cachem_1.1.0      yaml_2.3.10       parallel_4.5.1    tools_4.5.1      
[33] tzdb_0.5.0        httpuv_1.6.16     curl_7.0.0        vctrs_0.6.5      
[37] R6_2.6.1          lifecycle_1.0.4   git2r_0.36.2      stringr_1.5.2    
[41] bit_4.6.0         fs_1.6.6          vroom_1.6.5       pkgconfig_2.0.3  
[45] callr_3.7.6       pillar_1.11.1     bslib_0.9.0       later_1.4.4      
[49] glue_1.8.0        Rcpp_1.1.0        xfun_0.53         tibble_3.3.0     
[53] tidyselect_1.2.1  rstudioapi_0.17.1 knitr_1.50        htmltools_0.5.8.1
[57] rmarkdown_2.29    readr_2.1.5       compiler_4.5.1    getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Weverton Gomes da Costa, Pós-Doutorando, Departamento de
Estatística - Universidade Federal de Viçosa, <a
href="mailto:wevertonufv@gmail.com"
class="email">wevertonufv@gmail.com</a><a href="#fnref1"
class="footnote-back">↩︎</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
